### ISSUE TITLE
Windows always open image window with the same size

### ISSUE THREAD

**Issue by gcorriga** [guillep/pharo-core#10658]
**Bug description**
When opening an image using the Pharo 9 Windows VM, the world window has the same size regardless of the size it had when it was previously saved.

**To Reproduce**
1. Start new Pharo 9 image.
2. Change the size (or Maximize)
3. Save and Quit
4. Restart the same image

**Expected behavior**
The world window should have the same size it had when it was saved. It instead has a different size.

**Screenshots**
World window after opening:
![image](https://user-images.githubusercontent.com/7032660/145881677-12144dc2-5921-46f5-95ac-b1da52054137.png)

Resize window before Save and Quit:
![image](https://user-images.githubusercontent.com/7032660/145881762-22c67d74-26f3-4420-8503-a5ba33e9cd66.png)

Window size after reopening:
![image](https://user-images.githubusercontent.com/7032660/145881834-f7c351f7-de28-4a5d-87cd-d37a5a7b6cc2.png)

**Version information:**
 - OS: Windows
 - Version: 11
 - Pharo Version: Pharo9.0.0
Build information: Pharo-9.0.0+build.1573.sha.0e09d756fc99383cbb498565200d9e5e9841ce11 (64 Bit)

**Expected development cost**
I can help if someone points me to the right place.

**Additional context**
This also happens in Pharo 10 as of 13Dec2021

**Comment by pavel-krivanek** [comment]
This seems to be fixed

**Comment by gcorriga** [comment]
Hi, I can still reproduce the issue with Pharo 12. Should this issue be reopened?

**Comment by pavel-krivanek** [comment]
Hi, that is interesting because I do not have this issue but I tested it on Windows 10 (and not having 11 around). Can you try it with very fresh installation of the Launcher + Pharo 12 in Windows Sandbox over your Windows 11?

**Comment by astares** [comment]
Yes - can reproduce here (Pharo 12 on Windows 11). Pharo seems to keep the aspect ratio - but not the size and position.

**Comment by guillep** [comment]
Hi, I changed the issue title. Now the opening of windows is driven by pharo code and the VM is completely agnostic.
Windows are managed through OSWindow, which uses SDL2 behind the scenes.

It would be nice if people with a Windows box debug a bit the `OSWorldRenderer>>#doActivate` method to see where the problem comes from.

**Comment by gcorriga** [comment]
It doesn't appear that the problem is coming from `OSWorldRenderer>>doActivate` - I can verify that the `world worldState realWindowExtent` remains the same during the executing of that method.

However I've noticed that the `realWindowExtent` when the image starts up is exactly half of what it should be - so in my case instead of 1368@842 it is 684@421. The screen where the window is displayed has a 200% scaling factor and resolution 2736x1824. I suspect the issue might be with the scaling factor.

**Comment by gcorriga** [comment]
If I set the scaling factor of the screen to 100% then the window opens with a `realWindowExtent` of 1368@842 (as opposed to 2736@1684).

So the problem seems to be that somewhere after `OSWorldRenderer>>doActivate` is evaluated, something is halving both the x and y values of the world `worldState realWindowExtent`

**Comment by gcorriga** [comment]
With a bit more debugging, I now suspect the problem is in OSWorldRenderer>>#checkForNewScreenSize.
